### DISCUSSION TITLE
How to disabling adding  "id" field to existing notes

### DISCUSSION THREAD

**Discussion by Archie-2021** [epwalsh/obsidian.nvim#610] [Q&A]
Hey guys, 

Just started to use this plugin , with a fresh config in neovim, and I am liking it a lot. My only issue is that in default config. for some reason it adds a id field to notes even if they already exist (and not just created).

I know I can just remove the recall table which would start it altogether  and get rid of it, But I like the idea, it is just that to apply on existing notes, any setting or suggestions on how to achieve this?

Thanks

**Comment by Archie-2021** [comment]
This is ridiculous , I can't disable id creation thing even by passing an empty string to recall table. I use an id in my filenames by default, I don't need this plugin to add a separate id field to my notes on save. How can I stop this?

**Reply by epwalsh** [reply]
What is a "recall table"? Frontmatter? You can disable frontmatter management with a config setting.

**Reply by Archie-2021** [reply]
> What is a "recall table"? Frontmatter? You can disable frontmatter management with a config setting.

sorry , it is called 'callback' . I calls a bunch of functions and I thought maybe it is controlling these changes.

How I can change this frontmatter management? I can't see anything in the default config you put in the README.

**Comment by Archie-2021** [comment]
I still can' find where I can disable this automatic frontmatter thing in the config. Can someone please help?

**Reply by jcdiv47** [reply]
Try passing a custom `note_frontmatter_func` to config and remove `id` in the table below,

```lua
-- Optional, alternatively you can customize the frontmatter data.
  ---@return table
  note_frontmatter_func = function(note)
    -- Add the title of the note as an alias.
    if note.title then
      note:add_alias(note.title)
    end

    local out = { id = note.id, aliases = note.aliases, tags = note.tags }

    -- `note.metadata` contains any manually added fields in the frontmatter.
    -- So here we just make sure those fields are kept in the frontmatter.
    if note.metadata ~= nil and not vim.tbl_isempty(note.metadata) then
      for k, v in pairs(note.metadata) do
        out[k] = v
      end
    end

    return out
  end,
```

i.e.,

```lua
local out = { id = note.id, aliases = note.aliases, tags = note.tags }
```

becomes

```lua
local out = { aliases = note.aliases, tags = note.tags }
```

You can find the above default setting for `note_frontmatter_func` in [README](https://github.com/epwalsh/obsidian.nvim?tab=readme-ov-file#configuration-options). Hope it helps.

**Reply by Archie-2021** [reply]
> Try passing a custom `note_frontmatter_func` to config and remove `id` in the table below,
> 
> ```lua
> -- Optional, alternatively you can customize the frontmatter data.
>   ---@return table
>   note_frontmatter_func = function(note)
>     -- Add the title of the note as an alias.
>     if note.title then
>       note:add_alias(note.title)
>     end
> 
>     local out = { id = note.id, aliases = note.aliases, tags = note.tags }
> 
>     -- `note.metadata` contains any manually added fields in the frontmatter.
>     -- So here we just make sure those fields are kept in the frontmatter.
>     if note.metadata ~= nil and not vim.tbl_isempty(note.metadata) then
>       for k, v in pairs(note.metadata) do
>         out[k] = v
>       end
>     end
> 
>     return out
>   end,
> ```
> 
> i.e.,
> 
> ```lua
> local out = { id = note.id, aliases = note.aliases, tags = note.tags }
> ```
> 
> becomes
> 
> ```lua
> local out = { aliases = note.aliases, tags = note.tags }
> ```
> 
> You can find the above default setting for `note_frontmatter_func` in [README](https://github.com/epwalsh/obsidian.nvim?tab=readme-ov-file#configuration-options). Hope it helps.

Thanks , this get rid of id but it seems like it also removes id if there is such a field in the note.
I just want to stop it from messing with my frontmatter until I figure out how to use it. Is there a better way to stop it from editing stuff automatically? removing fields from the list just force delete them.

**Reply by jcdiv47** [reply]
Then I guess instead of removing `id = note.id` from table `out`, you could try something like:

```lua
id = note.metadata.id or note.id
```

**Reply by rodhash** [reply]
> note_frontmatter_func

For me this seems to be working just fine.. thank you.

**Comment by diego898** [comment]
Came here with same and related concern - I dont want to use zettelkasten at all:
- no id in title when a new note is created, and 
- no id in frontmatter of exisitng notes.

There should be a simple toggle to "disable_zettelkasten" or something similar.

**Comment by GitMurf** [comment]
Wondering the same thing. There is not much information about the concept of ID in the readme / docs. Is this required by the plugin? I would simply like to keep my notes the same as they are in native Obsidian where I control the frontmatter and nothing is added / removed "automatically".

It seems potentially the best solution right now is to use the frontmatter cb function to exclude ID from being written (like explained in this comment above: https://github.com/epwalsh/obsidian.nvim/discussions/610#discussioncomment-9702384)? Curious @epwalsh if this is the best recommended way to accomplish this?

Ultimately the question I have which I have not been able to find an answer to yet is whether the concept of a note ID is "required" in any other parts of the obsidian.nvim plugin or if everything will function normally even if we remove the ID field from all notes?

Thanks @epwalsh for your awesome work on this awesome plugin!!
